test/WebJobs.Script.Tests/Rpc/GenericWorkerProviderTests.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Microsoft.Azure.WebJobs.Script.Abstractions;
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Azure.WebJobs.Script.Rpc;
+using Microsoft.Extensions.Configuration;
+using Moq;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Rpc
+{
+    public class GenericWorkerProviderTests
+    {
+        [Fact]
+        public void Constructor_ThrowsNullArgument()
+        {
+            Assert.Throws<ArgumentNullException>(() => new GenericWorkerProvider(null, new List<string>()));
+            Assert.Throws<ArgumentNullException>(() => new GenericWorkerProvider(new WorkerDescription(), null));
+        }
+
+        [Fact]
+        public void GetDescription_ReturnsDescription()
+        {
+            var workerDescription = new WorkerDescription();
+            var arguments = new List<string>();
+            var provider = new GenericWorkerProvider(workerDescription, arguments);
+
+            Assert.Equal(workerDescription, provider.GetDescription());
+        }
+
+        [Fact]
+        public void TryConfigureArguments_ReturnsTrue()
+        {
+            var provider = new GenericWorkerProvider(new WorkerDescription(), new List<string>());
+            var args = new ArgumentsDescription();
+
+            Assert.True(provider.TryConfigureArguments(args, null, null));
+        }
+
+        [Fact]
+        public void ReadWorkerProviderFromConfig_ReturnsProvider()
+        {
+            string language = "test";
+            var description = new WorkerDescription()
+            {
+                DefaultExecutablePath = "foopath",
+                DefaultWorkerPath = "./src/index.test",
+                Language = "test",
+                Extension = ".test"
+            };
+
+            var arguments = new string[] { "-v", "verbose" };
+
+            string rootPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            string workerPath = Path.Combine(rootPath, language);
+            try
+            {
+                Directory.CreateDirectory(workerPath);
+
+                JObject config = new JObject();
```

Comment:
Add another test to verify we throw if config file is missing

Reply:
This shouldn't throw if config file is missing, but it will produce a critical log which I can test.